rust/opendp/src/trans/mod.rs

```diff
@@ -0,0 +1,246 @@
+//! Various implementations of Transformations.
+//!
+//! The different [`Transformation`] implementations in this module are accessed by calling the appropriate constructor function.
+//! Constructors are named in the form `make_xxx()`, where `xxx` indicates what the resulting `Transformation` does.
+
+use std::iter::Sum;
+use std::marker::PhantomData;
+use std::ops::{Bound, Mul, Sub, Div};
+
+use crate::core::{DatasetMetric, Domain, Metric, SensitivityMetric, Transformation};
+use crate::dist::{HammingDistance, L1Sensitivity, SymmetricDistance};
+use crate::dom::{AllDomain, IntervalDomain, VectorDomain, SizedDomain};
+pub use crate::trans::dataframe::*;
+use num::{Signed, NumCast};
+use std::cmp::Ordering;
+
+pub mod dataframe;
+
+// Trait for all constructors, can have different implementations depending on concrete types of Domains and/or Metrics
+pub trait MakeTransformation0<DI: Domain, DO: Domain, MI: Metric, MO: Metric> {
+    fn construct() -> crate::core::Transformation<DI, DO, MI, MO>;
+}
+
+pub trait MakeTransformation1<DI: Domain, DO: Domain, MI: Metric, MO: Metric, P1> {
+    fn construct(param1: P1) -> crate::core::Transformation<DI, DO, MI, MO>;
+}
+
+pub trait MakeTransformation2<DI: Domain, DO: Domain, MI: Metric, MO: Metric, P1, P2> {
+    fn construct(param1: P1, param2: P2) -> crate::core::Transformation<DI, DO, MI, MO>;
+}
+
+pub trait MakeTransformation3<DI: Domain, DO: Domain, MI: Metric, MO: Metric, P1, P2, P3> {
+    fn construct(param1: P1, param2: P2, param3: P3) -> crate::core::Transformation<DI, DO, MI, MO>;
+}
+
+pub trait MakeTransformation4<DI: Domain, DO: Domain, MI: Metric, MO: Metric, P1, P2, P3, P4> {
+    fn construct(param1: P1, param2: P2, param3: P3, param4: P4) -> crate::core::Transformation<DI, DO, MI, MO>;
+}
```

Comment:
Can we change the associated function name to make()? I think that's more in keeping with the terminology used in the paper.